recipes/laszip/all/conanfile.py

```diff
@@ -45,36 +37,37 @@ def configure(self):
         if self.options.shared:
             del self.options.fPIC
 
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
 
-    def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmake = self._configure_cmake()
-        cmake.build()
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["LASZIP_BUILD_STATIC"] = not self.options.shared
+        tc.generate()
 
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
+    def build(self):
+        apply_conandata_patches(self)
         cmake = CMake(self)
-        cmake.definitions["LASZIP_BUILD_STATIC"] = not self.options.shared
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+        cmake.configure()
+        cmake.build()
 
     def package(self):
-        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, "COPYING", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
 
     def package_info(self):
-        suffix = tools.Version(self.version).major if self.settings.os == "Windows" else ""
+        suffix = str(self.version).split(".")[0] if self.settings.os == "Windows" else ""
```

Comment:
if you request Conan >= 1.50.2, you can use conan.tools.scm.Version().major

Reply:
done